Fix: use new library to parse docx file and add tests

StrongMonkey opened the pull request:
The previous library we are using doesn't seem to be able to parse the text from docx. Switching to a new library that works. Also added test for that.

https://github.com/gptscript-ai/knowledge/issues/7

iwilltry42 reviewed (commented):
I didn't use this lib because it has external dependencies and often works with exec calls, e.g. https://github.com/sajari/docconv/blob/master/doc.go#L84

Is this not a problem here? 

Also, does this account for ODT still?

StrongMonkey commented:
@iwilltry42 For docx it will fallback to use https://github.com/sajari/docconv/blob/master/docx.go#L25 but I have switched to using convertDocx. For rtf it does require external binary so I've fallback to use the old library. Also addressed your comment about ODT, added seperate parser for Docx, RTF and ODT.

iwilltry42 commented on pkg/datastore/documentloader/defaults.go:
```diff
@@ -67,17 +67,42 @@ func DefaultDocLoaderFunc(filetype string) func(ctx context.Context, reader io.R
 			return FromGolc(golcdocloaders.NewNotebook(reader)).Load(ctx)
 		}
-	case ".docx", ".odt", ".rtf", "application/vnd.oasis.opendocument.text", "text/rtf", "application/vnd.openxmlformats-officedocument.wordprocessingml.document":
+	case ".docx", ".odt", ".rtf", "text/rtf", "application/vnd.oasis.opendocument.text", "application/vnd.openxmlformats-officedocument.wordprocessingml.document":
```
Do we actually support plain old `.doc`?

StrongMonkey replied:
I can test it out, I think it should be backward compatiable.

StrongMonkey replied:
It turns out that it could be pretty tricky to support doc. `.doc` is really a old format of doc and not many OSS libraries support it today. I think we can just say we don't support .doc and only newer format like `.docx` is supported. Even unidoc doesn't support it. https://help.unidoc.io/article/147-how-to-convert-ms-word-docx-file-into-pdf-using-unioffice
